### ISSUE TITLE
Truffle DSL processor fails with ClassCastException in 19.3.0 (regression)

### ISSUE THREAD

**Issue by fcurts** [oracle/graal#1933]
I'm trying to update my Truffle language from GraalVM 19.2.1 to 19.3.0, but compilation immediately fails with:

```
  java.lang.ClassCastException: com.sun.tools.javac.code.Type$ClassType cannot be cast to com.sun.tools.javac.code.Type$ArrayType
        at com.sun.tools.javac.code.Type$AnnotatedType.getComponentType(Type.java:2018)
        at com.oracle.truffle.dsl.processor.java.ElementUtils.isAssignableImpl(ElementUtils.java:476)
        at com.oracle.truffle.dsl.processor.java.ElementUtils.isAssignable(ElementUtils.java:396)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.persistCache(FlatNodeGenFactory.java:3924)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.persistAndInitializeCache(FlatNodeGenFactory.java:3879)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.initializeCaches(FlatNodeGenFactory.java:3864)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.createSpecialize(FlatNodeGenFactory.java:3311)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.visitSpecializationGroup(FlatNodeGenFactory.java:3025)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.visitSpecializationGroupChildren(FlatNodeGenFactory.java:3143)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.visitSpecializationGroup(FlatNodeGenFactory.java:2958)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.createExecuteAndSpecialize(FlatNodeGenFactory.java:1523)
        at com.oracle.truffle.dsl.processor.generator.FlatNodeGenFactory.create(FlatNodeGenFactory.java:495)
        at com.oracle.truffle.dsl.processor.generator.NodeCodeGenerator.generateNodes(NodeCodeGenerator.java:279)
        at com.oracle.truffle.dsl.processor.generator.NodeCodeGenerator.createImpl(NodeCodeGenerator.java:99)
        at com.oracle.truffle.dsl.processor.generator.NodeCodeGenerator.create(NodeCodeGenerator.java:81)
        at com.oracle.truffle.dsl.processor.generator.NodeCodeGenerator.create(NodeCodeGenerator.java:76)
        at com.oracle.truffle.dsl.processor.AnnotationProcessor.process(AnnotationProcessor.java:103)
        at com.oracle.truffle.dsl.processor.TruffleProcessor.processElement(TruffleProcessor.java:134)
        at com.oracle.truffle.dsl.processor.TruffleProcessor.processImpl(TruffleProcessor.java:121)
        at com.oracle.truffle.dsl.processor.TruffleProcessor.process(TruffleProcessor.java:85)
```

Googling the error message leads to this (from https://docs.oracle.com/javase/7/docs/api/javax/lang/model/type/TypeMirror.html):

```
To implement operations based on the class of an TypeMirror object, 
either use a visitor or use the result of the getKind() method.
Using instanceof is not necessarily a reliable idiom for determining the 
effective class of an object in this modeling hierarchy since an implementation 
may choose to have a single object implement multiple TypeMirror subinterfaces.
```

I'm compiling with this Java version:

```
java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)
```

**Comment by fcurts** [comment]
Will this be fixed in 20.0?

**Comment by boris-spas** [comment]
Tracking internally as Issue GR-20942.

**Comment by chumer** [comment]
Sorry for the delay. We are working on this now.

**Comment by tzezula** [comment]
Fixed in GraalVM 20.0.0 and dev by commit: https://github.com/oracle/graal/commit/93b02e7ab62731c74943e2aba557bd5cc810c4ec
